ERCS/Token with built-in AMM.md

```diff
@@ -0,0 +1,129 @@
+---
+eip:
```

Comment:
```suggestion
eip: 7850
```
Assigning next sequential EIP/ERC/RIP number.
Numbers are assigned by editors & associates.

Please also update the filename.